### PR TITLE
Use separate timers to control lift and tilt window coverings

### LINKED ISSUE #22287: [TC-WNCV-3.2] CurrentPositionLiftPercent100ths attribute read value mismatch error
Summary:
As per testplan of WNCV-3.2 in Step 3b: TH reads CurrentPositionLiftPercent100ths  and the expected outcome value must not be 0 ( 0x0 ) but got 0 when executed.
Steps needs review 3b , 3c, 3d, 3e
Client app: All clusters app
Server app: Chip-tool

Diff link:https://github.com/project-chip/connectedhomeip/compare/master...manjunath-grl:connectedhomeip:WNCV_3_2_Value_Mismatch

Testplan screenshot:
image

Execution logs:
[WNCV_3_2.txt](https://github.com/project-chip/connectedhomeip/files/9455969/WNCV_3_2.txt)

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Issue Being Resolved
* Fixes #22287 
* This is follow-up the review comments in https://github.com/project-chip/connectedhomeip/pull/22497. The spec requires we are able to control the lift and tilt window coverings individually since they are controlled by two different actuators in real product

#### Change overview
Use separate timers to control lift and tilt window coverings

### CODE REVIEW

File: examples/all-clusters-app/linux/WindowCoveringManager.cpp [hunk 1]
```diff
@@ -49,52 +49,86 @@ void WindowCoveringManager::HandleMovementTimer(System::Layer * layer, void * aA
...
-        if (currentPosition > targetPosition)
-        {
-            uint16_t tempPosition =
-                currentPosition > kDefaultMovementStep ? static_cast<uint16_t>(currentPosition - kDefaultMovementStep) : 0;
-            currentPosition = tempPosition > targetPosition ? tempPosition : targetPosition;
-
-            manager->mCurrentPosition.SetNonNull(currentPosition);
-        }
-        else
+        if (ComputeOperationalState(targetPosition, currentPosition) != OperationalState::MovingUpOrOpen)
```
Comment (jmeg-sfy) [inline examples/all-clusters-app/linux/WindowCoveringManager.cpp:56]: better to use this thanks !

File: examples/all-clusters-app/linux/WindowCoveringManager.cpp [hunk 2]
```diff
@@ -49,52 +49,86 @@ void WindowCoveringManager::HandleMovementTimer(System::Layer * layer, void * aA
...
         {
             ChipLogProgress(NotSpecified, "Reached the target position");
             return;
         }
+
+        uint16_t tempPosition =
```
Comment (jmeg-sfy) [inline examples/all-clusters-app/linux/WindowCoveringManager.cpp:62]: use this please
Percent100ths ComputePercent100thsStep(OperationalState direction, Percent100ths previous, Percent100ths delta);

File: examples/all-clusters-app/linux/WindowCoveringManager.cpp [hunk 3]
```diff
@@ -49,52 +49,86 @@ void WindowCoveringManager::HandleMovementTimer(System::Layer * layer, void * aA
...
-            manager->mCurrentPosition.SetNonNull(currentPosition);
-        }
-        else
+        if (ComputeOperationalState(targetPosition, currentPosition) != OperationalState::MovingDownOrClose)
         {
             ChipLogProgress(NotSpecified, "Reached the target position");
             return;
         }
+
+        uint16_t tempPosition = static_cast<uint16_t>(currentPosition + kDefaultMovementStep);
```
Comment (jmeg-sfy) [inline examples/all-clusters-app/linux/WindowCoveringManager.cpp:76]: please use this function for delta step
Percent100ths ComputePercent100thsStep(OperationalState direction, Percent100ths previous, Percent100ths delta);

File: examples/all-clusters-app/linux/WindowCoveringManager.cpp [hunk 4]
```diff
@@ -49,52 +49,86 @@ void WindowCoveringManager::HandleMovementTimer(System::Layer * layer, void * aA
...
+    if (OperationalState::MovingUpOrOpen == manager->mState)
+    {
+        if (ComputeOperationalState(targetPosition, currentPosition) != OperationalState::MovingUpOrOpen)
+        {
+            ChipLogProgress(NotSpecified, "Reached the target position");
+            return;
+        }
+
+        uint16_t tempPosition =
+            currentPosition > kDefaultMovementStep ? static_cast<uint16_t>(currentPosition - kDefaultMovementStep) : 0;
```
Comment (jmeg-sfy) [inline examples/all-clusters-app/linux/WindowCoveringManager.cpp:109]: step Percent100ths ComputePercent100thsStep(OperationalState direction, Percent100ths previous, Percent100ths delta);

**Review by jmeg-sfy — APPROVED** [review]
there is a function to compute step already , would be better to use it thanks

**Review by andy31415 — APPROVED** [review]

**Pull request merged**